### PR TITLE
Support Ruby 2.7.

### PR DESCRIPTION
**Pull request by tpendragon**
Refs #952

### CODE REVIEW

File: app/assets/stylesheets/blacklight.scss [hunk 1]
```diff
@@ -1,3 +1,16 @@
 @import 'bootstrap';
 
 @import 'blacklight/blacklight';
+
+.blacklight-icons svg {
```
Comment (hackartisan) [inline app/assets/stylesheets/blacklight.scss:5]: Were these removed upstream?

File: app/views/catalog/_search_within_header.html.erb [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
   <wrapper class="search-results-heading" maxWidth="1125">
     <h1 class="top-content-title">
-      <%= t(render_search_to_page_header(params)) %>
+      <%= render_search_to_page_header(params) %>
```
Comment (hackartisan) [inline app/views/catalog/_search_within_header.html.erb:5]: translation was moved into this helper I assume?

File: app/views/catalog/_show_main_content.html.erb [hunk 3]
```diff
@@ -1,17 +1,23 @@
-<% @page_title = t('blacklight.search.show.title', :document_title => document_show_html_title, :application_name => application_name).html_safe %>
+<% @page_title = t('blacklight.search.show.title', document_title: Deprecation.silence(Blacklight::BlacklightHelperBehavior) { document_show_html_title }, application_name: application_name).html_safe %>
```
Comment (hackartisan) [inline app/views/catalog/_show_main_content.html.erb:1]: Is there a ticket we need to go along with this deprecation silencing?

File: app/views/catalog/_show_main_content.html.erb [hunk 4]
```diff
@@ -1,17 +1,23 @@
...
 <% content_for(:head) { render_link_rel_alternates } %>
 
-<div id="document" class="document <%= render_document_class %>" itemscope  itemtype="<%= @document.itemtype %>">
-  <div id="doc_<%= @document.id.to_s.parameterize %>">
-    <%= render_document_partials @document, blacklight_config.view_config(:show).partials %>
-  </div>
-</div>
+<%= render (blacklight_config.view_config(:show).document_component || Blacklight::DocumentComponent).new(presenter: document_presenter(@document), component: :div, title_component: :h1, show: true) do |component| %>
```
Comment (hackartisan) [inline app/views/catalog/_show_main_content.html.erb:4]: If we know why we're overriding this file it would be nice to have a comment about it.

**Review by hackartisan — CHANGES_REQUESTED** [review]
I had a few questions if you please

File: app/assets/stylesheets/blacklight.scss (hunk 1 above)
Comment (tpendragon) [reply app/assets/stylesheets/blacklight.scss:5]: Yes. See https://github.com/projectblacklight/blacklight/pull/2225 and https://github.com/projectblacklight/blacklight/issues/2230

File: app/views/catalog/_search_within_header.html.erb (hunk 2 above)
Comment (tpendragon) [reply app/views/catalog/_search_within_header.html.erb:5]: We customized this and the customization was calling `t` when it shouldn't - I think this was probably broken before and relying on `t` falling back to whatever you passed to it if a key wasn't found.

File: app/views/catalog/_show_main_content.html.erb (hunk 3 above)
Comment (tpendragon) [reply app/views/catalog/_show_main_content.html.erb:1]: No, this is just me copying over the new `show_main_content` from blacklight.

File: app/views/catalog/_show_main_content.html.erb (hunk 4 above)
Comment (tpendragon) [reply app/views/catalog/_show_main_content.html.erb:5]: Done.

**Review by hackartisan — APPROVED** [review]
Thanks!

**Pull request merged**